requests-for-comments/2021-10-05-group-analytics.md

```diff
@@ -0,0 +1,196 @@
+
+To give an example:
+
+- Throughout insights, we’ll allow filtering insights with group properties. This will be implemented as a new tab per group type in the filters modal.
+- In trends, users can also select “active organizations” after first capturing events for an “organization” 
+- In funnels, users can select to analyze by a group type (e.g. “instance”). Different users within the group can have completed different steps of the funnel.
+- In retention, users can select “organization” over “user” for the unit to analyze retention by
+- And so on…
+
+
+### In-App: profile pages for groups
+
+We’ll also expose pages for different groups in-app, with:
+
+
+1. Listing all your different group types
+2. Listing and allowing searching all different groups of a type (e.g. organizations)
+3. Group page with
+    1. Group properties
+    2. Events from that group
+    3. Related users and groups (calculated via users’ other events)
+
+Initially, the design and behavior will be taken from the person pages and iterated later on.
+
+
+### Autocapture & posthog-js
+
+To make autocapture work with groups and to make posthog-js usage a bit simpler, we can provide the following API:
+
+```js
+    // Normal posthog usage
+    posthog.identify(userId)
+    // New proposed method, to be called right after .identify
+    posthog.group('instance', 'posthog.yourapp.com', { instance_prop: 5})
+```
+
+After calling this method, we will automatically associate events from that session with that group until either:
+
+- `posthog.group` is called with the same group type
+- `posthog.reset()` is called.
+### Feature flags
+
+Users should be able to add group-level property filters to feature flags and be able to ensure rollouts at a group level (?).
+
+However feature flags will not be part of the initial scope of the project, see below.
+
+## Open questions
+### What pricing tier this belongs to
+
+Does this sound like a collaboration-related feature we should immediately upsell? This is what mixpanel and amplitude do.
+
+### Feature flags and posthog-js timing
+
+Feature flags should work with group filters eventually. One tricky part there is behavior of posthog-js and ensuring we “know” what groups are currently active at the time of the request.
+
+### How would “managing” groups go, if users manage to mess up?
+
+TBD, can focus on this after the initial MVP
+
+## Other details
+
+### Initial scope
+
+This project is quite large. Rather than update all of our client libraries, ingestion, queries, feature flags and insights pages at once, the plan would be to do a gradual rollout in phases, gathering feedback from stakeholders at each step.
+
+The first step would be an internal MVP, where we can experiment with the feature ourselves behind a feature flag.
+
+Some tasks involve implementing new logic in posthog-js and posthog-python client libraries, get queries working for trends/funnels, implement very simple group profile pages.
```

Comment:
It would be great if you could be a bit more specific about what you think will be possible / not possible with the MVP so we can make sure we can get the most value out of it to validate it effectively.

Would be keen to see if the MVP meets the basic needs of the growth team so we can validate on stuff we're working on right away, @kpthatsme @samwinslow?

Reply:
We'd continue working on the feature even after this sprint so want to avoid going too low level here - it's really not helpful. The rough order is:

1. Ingestion + js/py client library changes
2. Trends/funnels
3. Groups pages
4. Other insights modules

A bit later:
5. Other client libraries
6. Feature flags

Reply:
Slightly different suggestion on prioritization-

I feel like feature flag support should be moved up – it is core functionality and our solution today is a non starter for orgs that take testing seriously (since they can't guarantee the same experience for teams).

I'd suggest prioritizing feature flags after groups pages support. 

Re: other client libraries – I think the most important thing here is for us to create the HTTP api so folks can still roll their own solution if needed. There's a lot of work to do for client libraries in general.

Reply:
I'm excited about group analytics in general!

Eventually, I'd like to see what kinds of queries we'll be able to build — a powerful way to look at data would be by "unique groups", such as we can already do with unique users. I could see us using that a lot.